### ISSUE TITLE
[Bug]: Figure out why `TREFHT` regridded reference variable is not regridding on `main` with ESMF bilinear

### ISSUE THREAD

**Issue by tomvothecoder** [E3SM-Project/e3sm_diags#817]
### What happened?

In PR #815:

> This PR validates the replication of the slice flag feature added in PR #750 to fix #759.
> 
> I re-ran `ex1.py` and found that all `.nc` files produced are now within the relative tolerance.
> 
> For metrics, all but one is within the relative tolerance. This single metric is related to the regridded reference data for `"TREFHT"` with the `land` region. In [trefht.py](https://github.com/E3SM-Project/e3sm_diags/pull/815/files#diff-35f583b4eda8fd8effb76e44533f3d7b4daa5f6b80499a4bb0f8c8e2ef1b06a9), I compare the `.nc` file for the regridded reference data between xCDAT/xESMF and CDAT/ESMF. I found that CDAT/ESMF doesn't seem to actually regrid the data since the original and regridded reference variables are the same, as noted in this PR review [comment](https://github.com/E3SM-Project/e3sm_diags/pull/815#discussion_r1625034843). I think this is leading to the large diffs along the coast lines, as shown below.

| xCDAT         | CDAT     | Diff |
|--------------|-----------|------------|
|    ![debug_TREFHT_actual](https://github.com/E3SM-Project/e3sm_diags/assets/25624127/2e645736-0c16-4f21-bfa3-173f401bdc2c) | ![debug_TREFHT_expected](https://github.com/E3SM-Project/e3sm_diags/assets/25624127/3837f3b2-8808-40bd-acc8-69011a41cbd2)     |     ![debug_TREFHT_diff](https://github.com/E3SM-Project/e3sm_diags/assets/25624127/67fc6a68-a6d1-424c-9017-753f75f84614)

> In PR #794, I noticed how bilinear regridding with xCDAT/xESMF produces nan location mismatches when comparing land sea masks to CDAT/ESMF. I think we should move forward for now and debug this more later on in a separate PR.

### What did you expect to happen? Are there are possible answers you came across?

The reference `TREFHT` variable should be regridded since the `mv2.regrid` is called below:

https://github.com/E3SM-Project/e3sm_diags/blob/c8404ab0f5523fee80cda396357f307b7ad1ff5b/e3sm_diags/driver/utils/general.py#L246-L254

### Environment

`main` branch

**Comment by tomvothecoder** [comment]
This will be a non-issue after #902.
